### PR TITLE
pmd:ImmutableField, squid:S2325 - Immutable Field, private methods th…

### PR DESCRIPTION
**Pull request by m-ezzat**
This pull request is focused on resolving occurrences of Sonar rules pmd:ImmutableField - Immutable Field
squid:S2325 - "private" methods that don't access instance data should be "static"

You can find more information about the issues here:
https://dev.eclipse.org/sonar/coding_rules#q=pmd:ImmutableField
https://dev.eclipse.org/sonar/coding_rules#q=squid:S2325

Please let me know if you have any questions.

M-Ezzat

### CODE REVIEW

File: src/main/java/com/fewlaps/quitnowemailsuggester/EmailSuggester.java [hunk 1]
```diff
@@ -114,7 +114,7 @@ private String fixDomain(String email, String badDomain, String goodDomain) thro
         return email;
     }
 
-    private String fixDomainAndTld(String email, String badDomainAndTld, String goodDomainAndTld) {
+    private static String fixDomainAndTld(String email, String badDomainAndTld, String goodDomainAndTld) {
```
Comment (rocboronat) [inline src/main/java/com/fewlaps/quitnowemailsuggester/EmailSuggester.java]: @m-ezzat hi! Having static methods makes testing hard, 'cause it's difficult to mock it using mockito. What do you think about this? Thanks!

Comment (m-ezzat) [inline src/main/java/com/fewlaps/quitnowemailsuggester/EmailSuggester.java]: Hi @rocboronat this link may be helpful https://github.com/jayway/powermock/wiki/MockitoUsage#a-full-example-for-mocking-stubbing--verifying-static-method

Comment (rocboronat) [inline src/main/java/com/fewlaps/quitnowemailsuggester/EmailSuggester.java]: Hi! I know PowerMock but... it's not the elegant way of mocking a method :-)

**Comment by rocboronat** [comment]
Thank you @m-ezzat ! :-)

**Pull request merged**